collect_app/src/main/java/org/odk/collect/android/dynamicpreload/ExternalDataUtil.java

```diff
@@ -345,4 +342,30 @@ public static boolean isAnInteger(String value) {
             return false;
         }
     }
+
+    /**
+     * The config row is a special row in the choices worksheet that defines how
+     * choices are mapped from an external .csv file with list_name, name and label,
+     * where these correspond to columns in the .csv file.
+     */
+    private static boolean containsConfigurationChoice(List<SelectChoice> selectChoices) {
```

Comment:
The test and the name are backwards from each other! If you find a choice that has a non-integer value, you should return true and if you get all the way through without one then return false.

At the call site you want `!containsConfigurationChoice` -- throw an exception if there is no file configuration choice.

Reply:
Fixed.